glmnet/linear.py

```diff
@@ -219,7 +219,13 @@ def _fit(self, X, y, sample_weight):
 
         exclude_vars = 0
 
-        relative_penalties = np.ones(X.shape[1], dtype=np.float64, order='F')
+        try:
+            # numpy does not have a good way to check if a variable does not
+            # exist; so this try-except block is used
+            _ = relative_penalties.shape
+        except:
+            relative_penalties = np.ones(X.shape[1], dtype=np.float64,
+                                         order='F')
```

Comment:
Since the default value is `None` in the caller, this should work:
```python
if relative_penalties is None:
    relative_penalties = np.ones(X.shape[1], dtype=np.float64, order='F')
```